### ISSUE TITLE
[RFC] Should obi warn the user if there's a new version of obi available?

### ISSUE THREAD

**Issue by EchoAbstract** [Oblong/obi#83]
There's currently no way to force users to upgrade their instance of obi (specifically on linux).  It might be nice for obi to let people know they should update (however, this can get annoying, in the case of something like pip, where I need to wait for ubuntu to upgrade it for me).

As for how that could work? We could have a file in the root of the repo called "latest-version", and just fetch it from obi by requesting `https://raw.githubusercontent.com/Oblong/obi/master/latest-version` and reporting if there's a mismatch.  If we can't fetch the file, no need to warn.

This is **not urgent**

**Comment by sandover** [comment]
I appreciate the idea but I am not persuaded that we need to act on it

**Comment by tomjakubowski** [comment]
I strongly dislike when command line tools reach out to the internet "sometimes", and without warning. `brew` does this now, but they at least have a decent reason to (fielding tons of support tickets b/c someone hasn't updated in years). A tool shouldn't purposely get in the way of what the user wants it to do; `obi go wall` should do nothing but rsync the project, build it, and run it on the wall.

I would prefer a periodic nagging reminder to check for an update to this.

**Comment by EchoAbstract** [comment]
@tomjakubowski I agree, I actually don't like this behavior in programs, however, even with the nagging to check for updates, I see linux users as just not updating.

My main reason for filing this RFC is to make sure we think about how to handle trailing users if we have any backwards incompatible changes.

**Comment by sandover** [comment]
I'm thinking about times when our group needed a particular tool version, which we were reliant on homebrew to get -- recent examples being cmake, git, and git-lfs. 

- in the case of cmake, there was a new version which we did NOT want.  We had the option of using the homebrew pinning feature to avoid it
- in the case of git and git-lfs, there have been times when we needed the latest and greatest

The fact that those version negotiations were conducted "out of band" -- neither the tool nor homebrew nagged us to upgrade -- seemed OK.  We had to get information about (e.g.) git-lfs from the git-lfs project itself; the tool was silent.  

Is there a clearly better world if we do things differently?

**Comment by tomjakubowski** [comment]
How about in the project.yaml:

```
obi-version: '>= 4.2.0'
```

And you get a loud warning or error if the obi version doesn't satisfy that [semver requirement](https://pypi.python.org/pypi/semver).

**Comment by sandover** [comment]
Hey, Tom.  That gives us a way to make breaking changes to `project.yaml`, doesn't it?

**Comment by EchoAbstract** [comment]
I like that Tom. I think that's cleaner and removes my concerns.
